test/new-e2e/pkg/utils/clients/ssh.go

```diff
@@ -78,3 +80,80 @@ func ExecuteCommand(client *ssh.Client, command string) (string, error) {
 
 	return string(stdout), err
 }
+
+// CopyFile create a sftp session and copy a single file to the remote host through SSH
+func CopyFile(client *ssh.Client, src string, dst string) error {
+	sftpClient, err := sftp.NewClient(client)
+
+	if err != nil {
+		return err
+	}
+
+	if err := copyFile(sftpClient, src, dst); err != nil {
+		return err
+	}
+	return nil
+}
+
+// CopyFolder create a sftp session and copy a folder to remote host through SSH
+func CopyFolder(client *ssh.Client, srcFolder string, dstFolder string) error {
+
+	sftpClient, err := sftp.NewClient(client)
+	if err != nil {
+		return err
+	}
+
+	if err := sftpClient.MkdirAll(dstFolder); err != nil {
+		return err
+	}
+
+	if err := copyFolder(sftpClient, srcFolder, dstFolder); err != nil {
+		return err
+	}
+	return nil
+}
+
+func copyFolder(sftpClient *sftp.Client, srcFolder string, dstFolder string) error {
+
+	folderContent, err := os.ReadDir(srcFolder)
+	if err != nil {
+		return err
+	}
+
+	for _, d := range folderContent {
+		if !d.IsDir() {
+			copyFile(sftpClient, path.Join(srcFolder, d.Name()), path.Join(dstFolder, d.Name()))
+		} else {
+			err := sftpClient.Mkdir(path.Join(dstFolder, d.Name()))
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Have you considered calling `MkdirAll` as the first step of `copyFolder` ?